File: examples/python_dashboardv2/raspberry/logs.py
from grafana_foundation_sdk.builders.dashboardv2beta1 import Panel, QueryGroup, Target

from .common import default_logs, basic_loki_query


def errors_in_system_logs() -> Panel:
    return (
        Panel()
        .title("Errors in system logs")
        .visualization(default_logs())
        .data(
            QueryGroup()
            .target(Target().query(basic_loki_query('{level=~"err|crit|alert|emerg", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("A"))
            .target(Target().query(basic_loki_query('{filename=~"/var/log/syslog*|/var/log/messages*", job="integrations/raspberrypi-node", instance="$instance"} |~".+(?i)error(?-i).+"')).ref_id("B"))
        )
    )


def auth_logs() -> Panel:
    return (
        Panel()
        .title("Auth logs")
        .visualization(default_logs())
        .data(
            QueryGroup()
            .target(Target().query(basic_loki_query('{unit="ssh.service", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("A"))
            .target(Target().query(basic_loki_query('{filename=~"/var/log/auth.log|/var/log/secure", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("B"))
        )
    )


def kernel_logs() -> Panel:
    return (
        Panel()
        .title("Kernel logs")
        .visualization(default_logs())
        .data(
            QueryGroup()
            .target(Target().query(basic_loki_query('{transport="kernel", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("A"))
            .target(Target().query(basic_loki_query('{filename="/var/log/kern.log", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("B"))
        )
    )


def all_system_logs() -> Panel:
    return (
        Panel()
        .title("All system logs")
        .visualization(default_logs())
        .data(
            QueryGroup()
            .target(Target().query(basic_loki_query('{transport!="", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("A"))
            .target(Target().query(basic_loki_query('{filename!="", job="integrations/raspberrypi-node", instance="$instance"}')).ref_id("B"))
        )
    )

File: examples/_go_dashboardv2/logs.go
package main

import (
	dashboard "github.com/grafana/cog/generated/go/dashboardv2beta1"
)

func errorsInSystemLogs() *dashboard.PanelBuilder {
	return dashboard.NewPanelBuilder().
		Title("Errors in system logs").
		Visualization(defaultLogs()).
		Data(
			dashboard.NewQueryGroupBuilder().
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{level=~"err|crit|alert|emerg", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("A"),
				).
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{filename=~"/var/log/syslog*|/var/log/messages*", job="integrations/raspberrypi-node", instance="$instance"} |~".+(?i)error(?-i).+"`)).RefId("B"),
				),
		)
}

func authLogs() *dashboard.PanelBuilder {
	return dashboard.NewPanelBuilder().
		Title("Auth logs").
		Visualization(defaultLogs()).
		Data(
			dashboard.NewQueryGroupBuilder().
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{unit="ssh.service", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("A"),
				).
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{filename=~"/var/log/auth.log|/var/log/secure", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("B"),
				),
		)
}

func kernelLogs() *dashboard.PanelBuilder {
	return dashboard.NewPanelBuilder().
		Title("Kernel logs").
		Visualization(defaultLogs()).
		Data(
			dashboard.NewQueryGroupBuilder().
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{transport="kernel", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("A"),
				).
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{filename="/var/log/kern.log", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("B"),
				),
		)
}

func allSystemLogs() *dashboard.PanelBuilder {
	return dashboard.NewPanelBuilder().
		Title("All system logs").
		Visualization(defaultLogs()).
		Data(
			dashboard.NewQueryGroupBuilder().
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{transport!="", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("A"),
				).
				Target(
					dashboard.NewTargetBuilder().Query(basicLokiQuery(`{filename!="", job="integrations/raspberrypi-node", instance="$instance"}`)).RefId("B"),
				),
		)
}
